### PR TITLE
Integrate the functionalities of create, update and delete for semesters

### PR DESCRIPTION
**Pull request by kusaljayawardhana**

### CODE REVIEW

File: resources/views/backend/semesters/index-table-row.blade.php [hunk 1]
```diff
@@ -7,7 +7,7 @@
 </x-livewire-tables::table.cell>
 
 <x-livewire-tables::table.cell>
-    @if ($row->academic_program == 0)
+    @if ($row->academic_program === 'Undergraduate')
```
Comment (NuwanJ) [inline resources/views/backend/semesters/index-table-row.blade.php:10]: Can't you use the `$row->academic_program` directly in here?

File: app/Domains/Semester/Models/Semester.php
Comment (NuwanJ) [inline app/Domains/Semester/Models/Semester.php:1]: Why were `getAcademicPrograms()` and other functions removed from here?

**Review by NuwanJ — CHANGES_REQUESTED** [review]
Can you check on the review comments?

File: app/Domains/Semester/Models/Semester.php
Comment (kusaljayawardhana) [reply app/Domains/Semester/Models/Semester.php:1]: They were duplicates

**Comment by kusaljayawardhana** [comment]
> Can you check on the review comments?

I will make the changes.

**Pull request merged**